consensus/core/src/synchronizer.rs

```diff
@@ -833,8 +981,9 @@ mod tests {
     // TODO: create a complete Mock for thread dispatcher to be used from several tests
     #[derive(Default)]
     struct MockCoreThreadDispatcher {
-        add_blocks: tokio::sync::Mutex<Vec<VerifiedBlock>>,
-        missing_blocks: tokio::sync::Mutex<BTreeSet<BlockRef>>,
+        add_blocks: Mutex<Vec<VerifiedBlock>>,
+        missing_blocks: Mutex<BTreeSet<BlockRef>>,
+        last_known_proposed_round: parking_lot::Mutex<Vec<Round>>,
```

Comment:
Why does this have to be Vec<Round>?

Reply:
Because that allow us to keep the whole call history . That's particularly useful to understand how many times the corresponding method has been called.